### ISSUE TITLE
Searching for samples on the website feels quite unpolished

### ISSUE THREAD

**Issue by Pandicon** [DevSnowflake/code-examples#318]
When looking through the samples on the website it didn't feel very good. The main cause of this is the need to include characters like - and _ when searching for some sample (like a hello world file). I think that ignoring these characters, which are in the file names just to not have spaces there, while searching would make the site much better.
Something else I noticed is that you can't search for language specific samples. This would be great for the languages that have a lot of samples, for example if I try to search for a hello world file in JS, I will either have to scroll through a lot of hello world files or through a lot of other JS samples.

https://user-images.githubusercontent.com/70060103/139596201-7dd5e939-25cb-4662-9ead-1371eb90d9c9.mp4

**Comment by rhygg** [comment]
The reason, you're finding this issue is because we aren't using a external library for the ``search algorithm``. 
The queries are interpreted by the **name** of the file.
Thus, to fix the issue we have to create our own search algorithm or use a external library.
I wonder if anyone has suggestions, would love to hear them 😄 !

We could totally implement [fuzzy-search](https://www.npmjs.com/package/fuzzy-search), since it's lightweight and works on zero-dependencies.

**Comment by DevAndromeda** [comment]
imo we should rebuild the website

**Comment by rhygg** [comment]
> imo we should rebuild the website

that's going to be a hefty job. The website looks fine, the aesthetics match the theme of the organization. It's better just to stick with this. Re-designing would take a good amount of time. We just need to go about fixing those minor bugs, in my opinion. 

tl;dr your choice.

**Comment by DevAndromeda** [comment]
this should now be fixed by 4cd6654a26b588ec5e9f8bd25e14f9dd8c74424b
